src/main/scala/io/iohk/ethereum/network/PeersInfoHolderActor.scala

```diff
@@ -0,0 +1,204 @@
+package io.iohk.ethereum.network
+
+import akka.actor.{Actor, ActorLogging, ActorRef, Props}
+import io.iohk.ethereum.db.storage.AppStateStorage
+import io.iohk.ethereum.network.PeerActor.{DisconnectPeer, SendMessage}
+import io.iohk.ethereum.network.PeersInfoHolderActor._
+import io.iohk.ethereum.network.PeerEventBusActor.{PeerSelector, Subscribe, Unsubscribe}
+import io.iohk.ethereum.network.PeerEventBusActor.PeerEvent._
+import io.iohk.ethereum.network.PeerEventBusActor.SubscriptionClassifier._
+import io.iohk.ethereum.network.handshaker.Handshaker.HandshakeResult
+import io.iohk.ethereum.network.p2p.Message
+import io.iohk.ethereum.network.p2p.messages.CommonMessages.{NewBlock, Status}
+import io.iohk.ethereum.network.p2p.messages.PV62.{BlockHeaders, GetBlockHeaders, NewBlockHashes}
+import io.iohk.ethereum.network.p2p.messages.WireProtocol.Disconnect
+
+class PeersInfoHolderActor(peerEventBusActor: ActorRef, appStateStorage: AppStateStorage,
+                           forkResolverOpt: Option[ForkResolver]) extends Actor with ActorLogging {
+
+  private type PeersWithInfo = Map[PeerId, PeerWithInfo]
+
+  peerEventBusActor ! Subscribe(PeerHandshaked)
+
+  override def receive: Receive = handlePeersInfoEvents(Map.empty)
+
+  def handlePeersInfoEvents(peersWithInfo: PeersWithInfo): Receive = {
+
+    case MessageToPeer(message, peerId) if peersWithInfo.contains(peerId) =>
+      val newPeersWithInfo = updatePeersWithInfo(peersWithInfo, peerId, message, handleSentMessage)
+      context become handlePeersInfoEvents(newPeersWithInfo)
+
+    case MessageFromPeer(message, peerId) if peersWithInfo.contains(peerId) =>
+      val newPeersWithInfo = updatePeersWithInfo(peersWithInfo, peerId, message, handleReceivedMessage)
+      context become handlePeersInfoEvents(newPeersWithInfo)
+
+    case PeerHandshakeSuccessful(peer, peerInfo: PeerInfo) =>
+      peerEventBusActor ! Subscribe(PeerDisconnectedClassifier(PeerSelector.WithId(peer.id)))
+      peerEventBusActor ! Subscribe(MessageReceivedClassifier(msgCodesWithInfo, PeerSelector.WithId(peer.id)))
+      peerEventBusActor ! Subscribe(MessageSentClassifier(msgCodesWithInfo, PeerSelector.WithId(peer.id)))
+
+      //Ask for the highest block from the peer
+      peer.ref ! SendMessage(GetBlockHeaders(Right(peerInfo.remoteStatus.bestHash), 1, 0, false))
```

Comment:
Should this actor ask for this? Isn't that a logic that should be within the one that does the handshaking?